cmd/kubelet/app/options/options.go

```diff
@@ -421,7 +421,10 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.BoolVar(&c.RotateCertificates, "rotate-certificates", c.RotateCertificates, "Auto rotate the kubelet client certificates by requesting new certificates from the kube-apiserver when the certificate expiration approaches.")
 
 	fs.Int32Var(&c.RegistryPullQPS, "registry-qps", c.RegistryPullQPS, "If > 0, limit registry pull QPS to this value.  If 0, unlimited.")
+	fs.MarkDeprecated("registry-qps", "will be removed in 1.32 or later. Use MaxParallelImagePulls instead.") //nolint:errcheck
 	fs.Int32Var(&c.RegistryBurst, "registry-burst", c.RegistryBurst, "Maximum size of a bursty pulls, temporarily allows pulls to burst to this number, while still not exceeding registry-qps. Only used if --registry-qps > 0")
+	fs.MarkDeprecated("registry-burst", "will be removed in 1.32 or later. Use MaxParallelImagePulls instead.") //nolint:errcheck
```

Comment:
presumably, these flags were GA, so we shouldn't deprecate them in favor of a beta feature, right? xref https://github.com/kubernetes/website/pull/42031

Reply:
good point @liggitt let's wait for @SergeyKanzhelev. we should land that policy change for 1.30 for sure!

Reply:
It sounds like we could deprecate it later when the MaxParallelImagePull is GA.

/hold

As the original flag doesn't work like expected, I still think we should deprecate it early.(And we can also show the stage of the replacements flag if the replacement is beta.) hold as we have the rule.